specification/containerinstance/resource-manager/Microsoft.ContainerInstance/stable/2018-04-01/containerInstance.json

```diff
@@ -0,0 +1,1234 @@
+              "type": "string"
+            },
+            "resource": {
+              "type": "string",
+              "description": "The name of the resource type of the operation."
+            },
+            "operation": {
+              "description": "The friendly name of the operation.",
+              "type": "string"
+            },
+            "description": {
+              "description": "The description of the operation.",
+              "type": "string"
+            }
+          }
+        },
+        "origin": {
+          "type": "string",
+          "description": "The intended executor of the operation.",
+          "enum": [
+            "User",
+            "System"
+          ],
+          "x-ms-enum": {
+            "name": "ContainerInstanceOperationsOrigin",
+            "modelAsString": true
+          }
+        }
+      },
+      "required": [
+          "name",
+          "display"
+      ]
+    },
+    "UsageListResult": {
+      "description": "The response containing the usage data",
+      "type": "object",
+      "properties": {
+        "value": {
+          "readOnly": true,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Usage"
+          }
+        }
+      }
+    },
+    "Usage": {
+      "description": "A single usage result",
+      "type": "object",
+      "properties": {
+        "unit": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Unit of the usage result"
+        },
+        "currentValue": {
+          "readOnly": true,
+          "type": "integer",
+          "description": "The current usage of the resource"
+        },
+        "limit": {
+          "readOnly": true,
+          "type": "integer",
+          "description": "The maximum permitted usage of the resource."
+        },
+        "name": {
+          "readOnly": true,
+          "type": "object",
+          "description": "The name object of the resource",
+          "properties": {
+            "value":{
+              "readOnly": true,
+              "type": "string",
+              "description": "The name of the resource"
+            },
+            "localizedValue": {
+              "readOnly": true,
+              "type": "string",
+              "description": "The localized name of the resource"
+            }
+          }
+        }
+      }
+    },
+    "ContainerGroupListResult": {
+      "description": "The container group list response that contains the container group properties.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ContainerGroup"
+          },
+          "description": "The list of container groups."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "The URI to fetch the next page of container groups."
+        }
+      }
+    },
+    "Logs": {
```

Comment:
>   "Logs": { [](start = 1, length = 12)

This type is used only as return value from the server. It is not supposed to be sent back to server thus all the properties should be marked as read-only.

Reply:
Done